File: packages/control-plane/test/fixtures/index/sample.go
// fixture：Go（struct / interface / 别名 / 带接收者的方法）。

package sample

const MaxRetries = 3

type Store struct {
	Name string
}

type Reader interface {
	Read(key string) (string, error)
}

type Key string

func NewStore(name string) *Store {
	return &Store{Name: name}
}

func (s *Store) Read(key string) (string, error) {
	return s.Name + key, nil
}

func (s *Store) Close() error {
	return nil
}

File: packages/control-plane/test/fixtures/index/sample.java
// fixture：Java（class / interface / enum / 构造器 / 方法；字段刻意不收）。

package sample;

import java.util.List;

public class Store {
    private final String name;

    public Store(String name) {
        this.name = name;
    }

    public String read(String key, List<String> extra) {
        return this.name + key + extra.size();
    }
}

interface Reader {
    String read(String key);
}

enum Mode {
    FAST,
    SLOW,
}
